Project/Source/Components/ComponentBilboardRender.cpp

```diff
@@ -0,0 +1,209 @@
+		ImGui::SameLine();
+		ImGui::TextWrapped("%i x %i", width, height);
+		ImGui::Image((void*) textureResource->glTexture, ImVec2(200, 200));
+		ImGui::InputScalar("Xtiles: ", ImGuiDataType_U32, &Xtiles);
+		ImGui::InputScalar("Ytiles: ", ImGuiDataType_U32, &Ytiles);
+		ImGui::InputFloat("Animation Speed: ", &animationSpeed);
+
+		ImGui::ColorEdit3("InitColor##", initC.ptr());
+		ImGui::ColorEdit3("FinalColor##", finalC.ptr());
+	}
+}
+
+void ComponentBilboardRender::Load(JsonValue jComponent) {
+	shaderID = jComponent[JSON_TAG_TEXTURE_SHADERID];
+
+	if (shaderID != 0) {
+		App->resources->IncreaseReferenceCount(shaderID);
+	}
+
+	textureID = jComponent[JSON_TAG_TEXTURE_TEXTUREID];
+
+	if (textureID != 0) {
+		App->resources->IncreaseReferenceCount(textureID);
+	}
+	Ytiles = jComponent[JSON_TAG_YTILES];
+	Xtiles = jComponent[JSON_TAG_XTILES];
+	animationSpeed = jComponent[JSON_TAG_ANIMATIONSPEED];
+
+	JsonValue jColor = jComponent[JSON_TAG_INITCOLOR];
+	initC.Set(jColor[0], jColor[1], jColor[2]);
+
+	JsonValue jColor2 = jComponent[JSON_TAG_FINALCOLOR];
+	finalC.Set(jColor2[0], jColor2[1], jColor2[2]);
+}
+
+void ComponentBilboardRender::Save(JsonValue jComponent) const {
+	jComponent[JSON_TAG_TEXTURE_SHADERID] = shaderID;
+	jComponent[JSON_TAG_TEXTURE_TEXTUREID] = textureID;
+	jComponent[JSON_TAG_YTILES] = Ytiles;
+	jComponent[JSON_TAG_XTILES] = Xtiles;
+	jComponent[JSON_TAG_ANIMATIONSPEED] = animationSpeed;
+
+	JsonValue jColor = jComponent[JSON_TAG_INITCOLOR];
+	jColor[0] = initC.x;
+	jColor[1] = initC.y;
+	jColor[2] = initC.z;
+	JsonValue jColor2 = jComponent[JSON_TAG_FINALCOLOR];
+	jColor2[0] = finalC.x;
+	jColor2[1] = finalC.y;
+	jColor2[2] = finalC.z;
+}
+
+void ComponentBilboardRender::Draw() {
+	unsigned int program = 0;
+	ResourceShader* shaderResouce = App->resources->GetResource<ResourceShader>(shaderID);
+	if (shaderResouce) {
+		program = shaderResouce->GetShaderProgram();
+	} else {
+		return;
+	}
+
+	glDepthMask(GL_FALSE);
+	glEnable(GL_BLEND);
+	glBlendEquation(GL_FUNC_ADD);
+	glBlendFunc(GL_ONE, GL_ONE);
+
+	glBindBuffer(GL_ARRAY_BUFFER, App->userInterface->GetQuadVBO());
+	glEnableVertexAttribArray(0);
+	glEnableVertexAttribArray(1);
+	glVertexAttribPointer(0, 3, GL_FLOAT, GL_FALSE, 0, (void*) 0);
+	glVertexAttribPointer(1, 2, GL_FLOAT, GL_FALSE, 0, (void*) (sizeof(float) * 6 * 3));
+	glUseProgram(program);
+
+	ComponentTransform* transform = GetOwner().GetComponent<ComponentTransform>();
+
+	float3x3 rotatePart = transform->GetGlobalMatrix().RotatePart();
+	Frustum* frustum = App->camera->GetActiveCamera()->GetFrustum();
+	float4x4* proj = &App->camera->GetProjectionMatrix();
+	float4x4* view = &App->camera->GetViewMatrix();
+
+	float4x4 newModelMatrix = transform->GetGlobalMatrix();
+	newModelMatrix = newModelMatrix.LookAt(rotatePart.Col(2), -frustum->Front(), rotatePart.Col(1), float3::unitY);
+	float4x4 Final = float4x4::FromTRS(transform->GetGlobalPosition(), newModelMatrix.RotatePart(), transform->GetGlobalScale());
```

Comment:
- I would put modelMatrix instead of newModelMatrix, and newModelMatrix instead of Final

- Remove the commented line